lib/project_types/script/layers/infrastructure/assemblyscript_dependency_manager.rb

```diff
@@ -32,6 +32,9 @@ def write_npmrc
           @ctx.system(
             'npm', '--userconfig', './.npmrc', 'config', 'set', '@shopify:registry', 'https://registry.npmjs.com'
           )
+          @ctx.system(
+            'npm', '--userconfig', './.npmrc', 'config', 'set', 'engine-strict', 'true'
```

Comment:
In the future we may want to get a confirm from the user to change their system config. This is bound to change someones config in a way that they have set otherwise and then they will open issues here for breaking their system.

Reply:
It is not a system config, it's updating a local `.npmrc` specific to their script.